server/task/src/test/java/org/apache/james/task/MemoryTaskManagerWorkerTest.java

```diff
@@ -0,0 +1,154 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.task;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.junit.Test;
+
+import reactor.core.publisher.Mono;
+
+public class MemoryTaskManagerWorkerTest {
+
+    private final MemoryTaskManagerWorker worker = new MemoryTaskManagerWorker();
+
+    private final Task successfulTask = () -> Task.Result.COMPLETED;
+    private final Task failedTask = () -> Task.Result.PARTIAL;
+    private final Task throwingTask = () -> {
+        throw new RuntimeException("Throwing Task");
+    };
+
+    @Test
+    public void aSuccessfullTaskShouldCompleteSuccessfully() {
+        assertTaskExecutionResult(successfulTask, Task.Result.COMPLETED, TaskManager.Status.COMPLETED);
```

Comment:
I don't know how to read such an assertion. 3 parameters are too much, and method extraction is more likely overkill here.